typescript/nomad-deploy/src/core/index.ts

```diff
@@ -201,10 +200,13 @@ export async function deployUnenrolledReplica(
     ? contracts.TestReplica__factory
     : contracts.Replica__factory;
 
+  // deploy the new Replica with the remote Home's current committedRoot
+  const currentRoot = await remote.contracts.home?.proxy.committedRoot();
+
```

Comment:
Don't we want to get committedRoot from events instead for 5-10 update lag?

Reply:
we are going with a simplified approach. consider the case when it's an initial deploy and there are no updates. there are too many edge cases with querying for events, and this works well as an alternative

Reply:
probably file a low-priority issue to improve this by going backwards a bit. right now it isn't very important. verrrrry edge case